src/libs/Navigation/NavigationRoot.js

```diff
@@ -1,115 +1,52 @@
 import {navigationRef} from './Navigation';
 import linkingConfig from './linkingConfig';
 import AppNavigator from './AppNavigator';
-import getPathName from './getPathName';
-import ONYXKEYS from '../../ONYXKEYS';
 import ROUTES from '../../ROUTES';
-import styles from '../../styles/styles';
-import themeColors from '../../styles/themes/default';
 import {updateCurrentlyViewedReportID} from '../actions/Report';
 import {setCurrentURL} from '../actions/App';
+import FullScreenLoadingIndicator from '../../components/FullscreenLoadingIndicator';
 
 const propTypes = {
     // Whether the current user is logged in with an authToken
     authenticated: PropTypes.bool.isRequired,
-
-    // The current reportID that we are navigated to or should show in the ReportScreen
-    currentlyViewedReportID: PropTypes.string,
-};
-
-const defaultProps = {
-    currentlyViewedReportID: null,
 };
 
 class NavigationRoot extends Component {
     constructor(props) {
         super(props);
 
-        this.state = {
-            loading: true,
-            initialState: undefined,
-        };
+        this.handleStateChange = this.handleStateChange.bind(this);
     }
 
-    componentDidMount() {
-        Linking.getInitialURL()
-            .then((initialUrl) => {
-                // On web we should be able to parse this. It will be null on native for now until deep links are
-                // hooked up
-                const path = getPathName(initialUrl);
-                let initialState = getStateFromPath(path, linkingConfig.config);
-                setCurrentURL(path);
-
-                // If we are landing on something other than the report screen or site root then we MUST set the
-                // initial route to the currently viewed report so there some history to navigate back from
-                if (path !== `/${ROUTES.HOME}` && !path.includes(`/${ROUTES.REPORT}`)) {
-                    const homeRoute = {
-                        name: 'Home',
-                    };
-
-                    if (this.props.currentlyViewedReportID) {
-                        homeRoute.params = {
-                            screen: 'Report',
-                            params: {
-                                reportID: this.props.currentlyViewedReportID,
-                            },
-                        };
-                    }
-
-                    if (!initialState) {
-                        initialState = {
-                            routes: [],
-                        };
-                    }
+    /**
+     * Intercept state changes and perform different logic
+     * @param {NavigationState} state
+     */
+    handleStateChange(state) {
+        if (!state) {
+            return;
+        }
 
-                    initialState.routes = [
-                        homeRoute,
-                        ...initialState.routes,
-                    ];
-                }
```

Comment:
This was causing 2 home routes/states to be stored in the initial state, though I didn't see any negative effects resulting from this.

Reply:
Nice. I think it might have been left behind from an earlier iteration and agree it makes no sense.

Reply:
And then I saw some negative effects 😃